src/box/index.h

```diff
@@ -575,6 +575,13 @@ struct index_vtab {
 					    const char *key,
 					    uint32_t part_count,
 					    const char *pos);
+	/** The same as create_iterator but skip first @offset tuples. */
+	struct iterator *(*create_iterator_with_offset)(struct index *index,
+							enum iterator_type type,
+							const char *key,
+							uint32_t part_count,
+							const char *pos,
+							uint32_t offset);
```

Comment:
Seems like a way to go if we don't want the slice check in the index virtual methods. The only alternative discussed in the RFC was introducing a new `uint32_t offset` parameter to the existing method.

I'd implement the way you suggested if @locker has no objections.

Reply:
> Let's add a new argument `uint32_t *offset` to `create_iterator` and each index will decrease `offset` value if it can do it better than N advancements. For example, tree index will always return 0 here since it can do offset in `O(log n)`, and hash index won't touch this argument at all so we will advance the iterator manually in `box_select`.

This would obscure the function contract. I see nothing wrong in adding a new virtual method - it looks clearer to me. The situation with pagination is different because an index implementation either supports it or not so it either succeeds or raises an error.

Reply:
OK, we decided to check slices in index, so there is no need to implement offset for other indexes outside `create_index`.
What's about separate methods - I still don't like it, but you've already discussed it in RFC, so I resolve the comment.